### ISSUE TITLE
Cannot use service plan with SKU F1/D1 for windows web apps

### ISSUE THREAD

**Issue by alfosua** [hashicorp/terraform-provider-azurerm#16930]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.2.0

### AzureRM Provider Version

3.7.0

### Affected Resource(s)/Data Source(s)

azurerm_windows_web_app, azurerm_service_plan

### Terraform Configuration Files

```hcl
resource "azurerm_resource_group" "rg" {
  name = "rg"
  location = "eastus"
}

resource "azurerm_service_plan" "plan" {
  name = "plan"
  location = azurerm_resource_group.rg.location
  name = azurerm_resource_group.rg.name
  sku_name = "F1"
  os_type = "Windows"
}

resource "azurerm_windows_web_app" "app" {
  name = "app"
  location = azurerm_resource_group.rg.location
  name = azurerm_resource_group.rg.name
  service_plan_id = azurerm_service_plan.plan.id

  site_config {}
}
```

### Debug Output/Panic Output

```shell
╷
│ Error: creating Windows Web App: (Site Name "app" / Resource Group "rg"): web.AppsClient#CreateOrUpdate: Failure sending request: StatusCode=0 -- Original Error: autorest/azure: Service returned an error. Status=<nil> <nil>
│
│   with azurerm_windows_web_app.app,
│   on apps.tf line 14, in resource "azurerm_windows_web_app" "app":
│   99: resource "azurerm_windows_web_app" "app" {
│
│ creating Windows Web App: (Site Name "app" / Resource Group "rg"): web.AppsClient#CreateOrUpdate:      
│ Failure sending request: StatusCode=0 -- Original Error: autorest/azure: Service returned an error. Status=<nil> <nil>
```

### Expected Behaviour

Create successfully the target web app resource, including the service plan and azure resource group.

### Actual Behaviour

After only creating the azure resource group and service plan, it appears the given error message (with no more context) and the web app resource is not created. This happens whenever the plan has a SKU of F1 or D1.

### Important Factoids

Using

**Comment by xiaxyi** [comment]
@alfosua  should be the property `always_on`, I'll take a look and update the code accordingly. The free tier does not support `always_on` set to true

**Comment by alfosua** [comment]
> @alfosua should be the property `always_on`, I'll take a look and update the code accordingly. The free tier does not support `always_on` set to true

Great, thanks. I will be trying soon. I think there should be a better error message, I guess.

**Comment by xiaxyi** [comment]
@alfosua I think so, let me test to validate my assumption

**Comment by xiaxyi** [comment]
@alfosua can you try the latest released azurerm provider?

**Comment by SwethaVatsavayi5** [comment]
Any update on this? I am getting the same error message for SKU 'Y1' function consumption plan for windows web app and I have windows function app working fine on the same SKU. I tried on the latest released azurerm provider.

**Comment by SwethaVatsavayi5** [comment]
and I am getting below error message when updating to SKU F1/D1

updating Service Plan: (Serverfarm Name "" / Resource Group ""):
│ web.AppServicePlansClient#CreateOrUpdate: Failure sending request: StatusCode=400 -- Original Error: Code="BadRequest"      
│ Message="You tried creating a function app in the 'Shared' SKU. Function Apps are not supported in Free and Shared plans.   
│ Please choose a different plan." Details=[{"Message":"You tried creating a function app in the 'Shared' SKU. Function Apps  
│ are not supported in Free and Shared plans. Please choose a different
│ plan."},{"Code":"BadRequest"},{"ErrorEntity":{"Code":"BadRequest","ExtendedCode":"59919","Message":"You tried creating a    
│ function app in the 'Shared' SKU. Function Apps are not supported in Free and Shared plans. Please choose a different       
│ plan.","MessageTemplate":"You tried creating a function app in the '{0}' SKU. Function Apps are not supported in Free and   
│ Shared plans. Please choose a different plan.","Parameters":["Shared"]}}]

**Comment by alfosua** [comment]
Hi @xiaxyi 

Sorry for the late response...

I already tried the latest version of azurerm. Trying to create a webapp with a free SKU name and always_on flag set to true ends with a good error message:

```
Error: always_on cannot be set to true when using Free, F1, D1 Sku
│
│   with azurerm_windows_web_app.mainserv_app,
│   on main.tf line 69, in resource "azurerm_windows_web_app" "mainserv_app":
│   69: resource "azurerm_windows_web_app" "mainserv_app" {
│
│ always_on cannot be set to true when using Free, F1, D1 Sku
```

As you said, I can create webapps with free SKUs successfully if I set the always_on flag to false. So a lot of thanks for your help! :)

My last feedback would be to add the site_config prefix in the error message. Like: "site_config.always_on cannot be set to true when using Free, F1, D1 Sku". So it is clear the property can be found and set in that resource's block.

I will close this issue.
